curriculum/challenges/english/25-front-end-development/lab-rainbow-painting/66b1ec7e2bb25947ca579bf3.md

```diff
@@ -0,0 +1,522 @@
+assert(!!containerChild && containerChild.width === "145px" && containerChild.height === "145px");
+```
+
+You should use absolute positioning to center `#white` respect to its parent.
+
+```js
+const containerChild = new __helpers.CSSHelp(document).getStyle('#white');
+assert(!!containerChild);
+const val = '102.5px';
+const topOrBottom = containerChild.top === val || containerChild.bottom === val;
+const leftOrRight = containerChild.left === val || containerChild.right === val;
+assert(topOrBottom && leftOrRight);
+```
+
+You should target `#cover` and set its background color to `white`.
+
+```js
+const containerChild = new __helpers.CSSHelp(document).getStyle('#cover');
+assert(!!containerChild && containerChild.backgroundColor === 'white');
+```
+
+You should target `#cover` and set its width to the same width of the biggest `div` element, and its height to half of that value.
+
+```js
+const containerChild = new __helpers.CSSHelp(document).getStyle('#cover');
+const size = containerChild.width === '250px' && containerChild.height === '125px';
+assert(!!containerChild && size);
+```
+
+You should target `#cover` and set its position to cover the bottom half of the colored `div` elements.
+
+```js
+const containerChild = new __helpers.CSSHelp(document).getStyle('#cover');
+const leftOrRight = containerChild.left === '50px' || containerChild.right === '50px';
+const topOrBottom = containerChild.top === '175px' || containerChild.bottom === '50px';
+assert(!!containerChild && leftOrRight && topOrBottom);
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="utf-8">
+    <link rel="stylesheet" href="./styles.css">
+    <title>Rainbow</title>
+</head>
+
+<body>
+    
+</body>
+
+</html>
+```
+
+```css
+
+```
+
+# --solutions--
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="utf-8">
+    <link rel="stylesheet" href="./styles.css">
+    <title>Rainbow</title>
+</head>
+
+<body>
+    <main>
+        <div id="container">
+            <div id="red"></div>
+            <div id="orange"></div>
+            <div id="yellow"></div>
+            <div id="green"></div>
+            <div id="blue"></div>
+            <div id="indigo"></div>
+            <div id="violet"></div>
+            <div id="white"></div>
+            <div id="cover"></div>
+        </div>
+    </main>
+</body>
+
+</html>
+```
+
+```css
+* {
+    box-sizing: border-box;
+    margin: 0;
+    padding: 0;
+}
+
+div {
+    position: absolute;
+}
+
+div:not([id="cover"]) {
+    border-radius: 50%;
+}
+
+#container {
+    width: 350px;
+    height: 350px;
+    right: 0;
+    left: 0;
+    top: 0;
+    bottom: 0;
+    margin: auto;
+}
+
+#red {
+    background-color: red;
+    width: 250px;
+    height: 250px;
+    top: 50px;
+    left: 50px;
+}
+
+#orange {
+    background-color: orange;
+    width: 235px;
+    height: 235px;
+    left: 57.5px;
```

Comment:
There should be clues on how `57.5` for `left` and `top` are calculated. 
The same goes for other colours.

Reply:
I guess I can add an hint, but since I'm asking to center with respect to the parent the calculation is always the same: (size of parent - size of current element) / 2